src/main/java/org/codehaus/plexus/build/resources/DefaultResources.java

```diff
@@ -0,0 +1,142 @@
+import javax.inject.Singleton;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.codehaus.plexus.build.BuildContext;
+
+/**
+ * Default implementation of the Resources interface.
+ * <p>
+ * This implementation delegates to the BuildContext for compatibility with existing
+ * build infrastructure. It provides a transition path from the File-based API to the
+ * modern Path-based API.
+ * </p>
+ */
+@Named("default")
+@Singleton
+public class DefaultResources implements Resources {
+
+    private final BuildContext buildContext;
+
+    /**
+     * Creates a new DefaultResources instance.
+     *
+     * @param buildContext the BuildContext to which operations will be delegated
+     */
+    @Inject
+    public DefaultResources(BuildContext buildContext) {
+        this.buildContext = buildContext;
+    }
+
+    @Override
+    public boolean hasDelta(Path file) {
+        if (file == null) {
+            return false;
+        }
+        return buildContext.hasDelta(file.toFile());
+    }
+
+    @Override
+    public void refresh(Path file) {
+        if (file != null) {
+            buildContext.refresh(file.toFile());
+        }
+    }
+
+    @Override
+    public OutputStream newOutputStream(Path file) throws IOException {
+        if (file == null) {
+            throw new IllegalArgumentException("file cannot be null");
+        }
+        return buildContext.newFileOutputStream(file.toFile());
+    }
+
+    @Override
+    public OutputStream newOutputStream(Path file, boolean derived) throws IOException {
+        OutputStream outputStream = newOutputStream(file);
+        if (derived) {
+            // Mark the file as derived after creating the output stream.
+            // In the default implementation, markDerived is a no-op, so the timing doesn't matter.
+            // Custom implementations that track derived files should consider wrapping the stream
+            // to defer marking until successful close() to avoid marking files on failed writes.
+            markDerived(file);
+        }
+        return outputStream;
+    }
+
+    @Override
+    public boolean isUptodate(Path target, Path source) {
+        if (target == null || source == null) {
+            return false;
+        }
+        return buildContext.isUptodate(target.toFile(), source.toFile());
+    }
+
+    @Override
+    public Path getPath(String relpath) {
+        if (relpath == null) {
+            throw new IllegalArgumentException("relpath cannot be null");
+        }
+        // Note: The BuildContext API doesn't expose basedir directly.
+        // This default implementation resolves paths relative to the current working directory.
+        // Custom implementations (e.g., IDE integrations) should override this method
+        // to resolve against the actual build context basedir.
+        return java.nio.file.Paths.get(relpath);
```

Comment:
We should use `LegacySupport` here to get the current maven project and return the basedir then.